### PR TITLE
RATIS-1309. Clarify ApplyTransaction Javadoc

### PR DESCRIPTION
**Pull request by amaliujia**
https://issues.apache.org/jira/browse/RATIS-1309

### CODE REVIEW

File: ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java [hunk 1]
```diff
@@ -439,9 +439,10 @@ default FollowerEventApi followerEvent() {
   TransactionContext applyTransactionSerial(TransactionContext trx);
 
   /**
-   * Apply a committed log entry to the state machine. This method can be called concurrently with
-   * the other calls, and there is no guarantee that the calls will be ordered according to the
-   * log commit order.
+   * Apply a committed log entry to the state machine. This method is called sequentially in
+   * strict serial order that the transactions have been committed in the log. However,
+   * due to this call returning a future, depending on implementations, the order of applying
+   * entries to state machine is not guaranteed to be the same as the log commit order.
```
Comment (szetszwo) [inline ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java:445]: > However, due to this call returning a future, depending on implementations, the order of applying entries to state machine is not guaranteed to be the same as the log commit order.

How about we rewrite the sentence above as below?

> Note that this method, which returns a future, is asynchronous.  The state machine implementation may choose to apply the log entries in parallel.  In that case, the order of applying entries to state machine could possibly be different from the log commit order.

**Review by szetszwo — COMMENTED** [review]
@amaliujia , thanks a lot for making this change.  A comment inlined.

File: ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java (hunk 1 above)
Comment (amaliujia) [reply ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java]: Sure your proposal is better :)

**Comment by amaliujia** [comment]
The flaky test shouldn't be related to this PR since no code is changed.

**Review by szetszwo — APPROVED** [review]
+1

**Pull request merged**